src/app/meds-n-labels/activity-2/activity-2.component.ts

```diff
@@ -0,0 +1,196 @@
+            temporaryValue = array[currentIndex];
+            array[currentIndex] = array[randomIndex];
+            array[randomIndex] = temporaryValue;
+        }
+        return array;
+    }
+
+    getImgElement(card) {
+        return document.getElementById(card).getElementsByTagName('img')[0];
+    }
+
+    /**
+     * Flip card on click
+     * @param {Number} card The index of the clicked element
+     */
+    choose(card) {
+            if (this._clicks === 2) {
+                return;
+            } else {
+                this.getImgElement(card).src = this._baseImgPath + this._faces[card];
+                if (this._clicks === 0) {
+                    this._firstchoice = card;
+                    this._clicks = 1;
+                } else if (this._firstchoice !== card) {
+                    this._clicks = 2;
+                    this._secondchoice = card;
+                    this._obs = Observable.interval(500)
+                            .do(i => this.check());
+                    this._subscription = this._obs.subscribe();
+                }
+            }
+        }
+
+    /**
+     * Check if a match is made
+     */
+    check() {
+        this._subscription.unsubscribe();
+        this._clicks = 0;
+        if (this._faces[this._secondchoice] === this._faces[this._firstchoice]) {
+            const medName = this._faces[this._secondchoice]; // or this.faces[this._firstchoice]
+            const med = medName.substr(0, medName.lastIndexOf('.'));
+            for (let i = 0; i < DIAGNOSIS.length; i++) {
+                if (DIAGNOSIS[i].name === med) {
+                    this.successMatchAlert(DIAGNOSIS[i].desc);
+                    this.isMatchArr[this._secondchoice] = false;
+                    this.isMatchArr[this._firstchoice] = false;
+                    break;
+                }
+            }
+            this._matches++;
+            if (this._matches === 8) {
+                this.successAlert();
+                this._status = {stage: 3, activity: 2};
+                this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
+                this.activityComplete = true;
+            }
+        } else {
+            this.getImgElement(this._firstchoice).src = this.backcard;
+            this.getImgElement(this._secondchoice).src = this.backcard;
+            return;
+        }
+    }
+
+    /**
+     * Alert message on a successful match
+     * @param {String} medName The name of the medication
+     * @param {String} msg     The side effect message to be displayed
+     */
+    successMatchAlert(msg) {
+        swal(
+            'Congratulations!<br>You matched a pair',
+            msg,
+            'success'
+        );
+    }
+
+    /**
+     * Alert on completing the activity
+     */
+    successAlert() {
+        swal(
+            'Good job!',
+            'You completed this activity!',
+            'success'
+        );
+    }
+
+    checkProgress() {
```

Comment:
this is just duplicated code with https://github.com/systers/PC-Prep-Kit/pull/25/files#diff-78168c46d826955cc8506302e6bba148R75

i'd create a global function `checkProgress(stage, activity)` instead